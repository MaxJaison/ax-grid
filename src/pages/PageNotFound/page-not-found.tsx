import { Center, Divider, Flex, Heading, Text } from '@chakra-ui/react';

export const PageNotFound = () => {
    return (
        <Center bg="white" h="100vh" color="black" alignItems="center">
            <Flex align="center" justify="center" h="45px">
                <Heading as="h2" size="2xl" gap="10">
                    404
                </Heading>
                <Divider orientation="vertical" size="xs" borderColor="black" borderWidth="2px" mx="5" />
                <Text fontSize="3xl">Page Not Found.</Text>
            </Flex>
        </Center>
    );
};
