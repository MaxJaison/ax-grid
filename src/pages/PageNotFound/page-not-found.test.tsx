import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import { PageNotFound } from './page-not-found.tsx';

describe('PageNotFound Component', () => {
    it('renders the PageNotFound component', async () => {
        render(<PageNotFound />);

        expect(screen.getByText(/Page Not Found./i)).toBeInTheDocument();
    });

    it('create PageNotFound snapshot', async () => {
        const { asFragment } = render(<PageNotFound />);
        expect(asFragment()).toMatchSnapshot();
    });
});
