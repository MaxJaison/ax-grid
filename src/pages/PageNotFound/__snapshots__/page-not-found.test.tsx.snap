// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageNotFound Component > create PageNotFound snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-17bhemt"
  >
    <div
      class="css-1attr2v"
    >
      <h2
        class="chakra-heading css-yxom3w"
      >
        404
      </h2>
      <hr
        aria-orientation="vertical"
        class="chakra-divider css-erojza"
      />
      <p
        class="chakra-text css-1tapsqs"
      >
        Page Not Found.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
